File: tests/xcalablemp/global-view/gmove/F/gmove_G2L_2.f90
program gmove_G2L_2

  character*25 tname
  call gmove_G2L_1a1t_b(tname)
  call gmove_G2L_1a1t_bc(tname)

end program

subroutine gmove_G2L_1a1t_b(tname)

character(*) tname
integer :: i
integer,parameter :: n=8
integer a(n),b(n)
integer xmp_node_num
!$xmp nodes p(2)
!$xmp template tx(n)
!$xmp distribute tx(block) onto p
!$xmp align a(i) with tx(i)

!$xmp loop (i) on tx(i)
do i=1,n
  a(i)=i
end do

do i=1,n
  b(i)=0
end do

!$xmp gmove
b(1:n)=a(1:n)

ierr=0
do i=1,n
  ierr=ierr+abs(b(i)-i)
end do

!$xmp reduction (max:ierr)
tname="gmove_G2L_1a1t_b"
call chk_int(tname, ierr)

end subroutine

subroutine gmove_G2L_1a1t_bc(tname)

character(*) tname
integer :: i
integer,parameter :: n=8
integer a(n),b(n)
integer xmp_node_num
!$xmp nodes p(2)
!$xmp template tx(n)
!$xmp distribute tx(cyclic(2)) onto p
!$xmp align a(i) with tx(i)

!$xmp loop (i) on tx(i)
do i=1,n
  a(i)=i
end do

do i=1,n
  b(i)=0
end do

!$xmp gmove
b(2:5)=a(5:8)

ierr=0
do i=2,5
  ierr=ierr+abs(b(i)-(i+3))
end do

!$xmp reduction (max:ierr)
tname="gmove_G2L_1a1t_bc"
call chk_int(tname, ierr)

end subroutine

File: tests/xcalablemp/global-view/gmove/C/gmove_G2L_2.c
#define NAMELEN 25
#define N 8
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

extern int chk_int(char name[], int ierr);

void gmove_G2L_1a1t_bc(){

char name[NAMELEN]="gmove_G2L_1a1t_bc";
int n=N;
double a[n];
#pragma xmp nodes p(2)
#pragma xmp template tx(0:n-1)
#pragma xmp distribute tx(cyclic(2)) onto p
#pragma xmp align a[i] with tx(i)

 int i,/*j,*/ierr;
  double b[N],err;

#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    a[i]=i+1;
  }

  for(i=0;i<n;i++){
    b[i]=0;
  }

#pragma xmp gmove
  b[4:4]=a[0:4];

  err=0.0;
  for(i=4;i<8;i++){
    err=err+fabs(b[i]-(i-3));
  }

#pragma xmp reduction (MAX:err)
  ierr=err;
  chk_int(name, ierr);

}

int main(){

  gmove_G2L_1a1t_bc();

  return 0;

}
